src/index.ts

```diff
@@ -118,13 +118,17 @@ export function project<T = any>(options?: string | Options) {
 /**
  * Determine if the metadata server is currently available.
  */
+let cachedIsAvailableResponse: boolean | undefined;
 export async function isAvailable() {
+  if (cachedIsAvailableResponse !== undefined) {
+    return cachedIsAvailableResponse;
+  }
   try {
     // Attempt to read instance metadata. As configured, this will
     // retry 3 times if there is a valid response, and fail fast
     // if there is an ETIMEDOUT or ENOTFOUND error.
     await metadataAccessor('instance', undefined, 0, true);
-    return true;
+    return (cachedIsAvailableResponse = true);
```

Comment:
My personal preference is to never do this kind of things :)  I'd better make an assignment and `return` separately. It's a matter of personal style of course, feel free to do as you want to.

Reply:
👍fixed.